Core/include/eASTMagneticField.hh

```diff
@@ -0,0 +1,227 @@
+// ********************************************************************
+//
+// eASTMagneticField.hh
+//   Header file of eAST Magnetic Field class
+//
+// History
+//   June 22nd, 2021 : first implementation
+//
+// ********************************************************************
+
+#ifndef eASTMagneticField_h
+#define eASTMagneticField_h 1
+
+#include "G4MagneticField.hh"
+#include "G4GenericMessenger.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4UnitsTable.hh"
+
+#include <vector>
+
+class G4EquationOfMotion;
+class G4MagIntegratorStepper;
+class G4ChordFinder;
+class G4FieldManager;
+class G4PropagatorInField;
+
+class eASTMagneticFieldMap {
+
+  public:
+    eASTMagneticFieldMap(const G4String& name);
+    ~eASTMagneticFieldMap() = default;
+
+  private:
+
+    // Static maps to indices of neighboring points
+    static const int kLinearMap[8][3];
+    static const int kCubicMap[64][3];
+
+  private:
+
+    // Field name
+    G4String fName{""};
+
+  public:
+
+    // Load field map from filename
+    bool Load(const G4String& filename);
+
+    // Get field value at point
+    void GetFieldValue(const G4double point[4], G4double *field) const {
+      field[0] = 0;
+      field[1] = 0;
+      field[2] = 0;
+      AddFieldValue(point, field);
+    };
+
+    // Add field value at point
+    void AddFieldValue(const G4double point[4], G4double *field) const;
+
+    // Print field value at point
+    void PrintFieldValue(const G4ThreeVector& point);
```

Comment:
Due to the messenger attached to this function (and the other Print function which calls this) we cannot make this const.